subprojects/packagefiles/expat/meson.build

```diff
@@ -24,44 +24,27 @@ project(
   'c',
   license : 'MIT',
   version : '2.4.8',
-  meson_version : '>= 0.50.0',
+  meson_version : '>= 0.56.0',
 )
 
+libtool_ver = '1.8.8'
+
 config = configuration_data()
 cc = meson.get_compiler('c')
 
 # Define how much context to retain around the current parse point.
 config.set('XML_CONTEXT_BYTES', 1024)
 
-dep_libbsd = dependency('libbsd', required : get_option('use_libbsd'))
-if dep_libbsd.found()
-  config.set('HAVE_LIBBSD', true)
-endif
-
-if dep_libbsd.found()
-  header = 'bsd/stdlib.h'
-else
-  header = 'stdlib.h'
-endif
-
-if cc.has_header_symbol(header, 'arc4random_buf')
-  config.set('HAVE_ARC4RANDOM_BUF', true)
-elif cc.has_header_symbol(header, 'arc4random')
-  config.set('HAVE_ARC4RANDOM', true)
-endif
+dep_libbsd = dependency('libbsd-overlay', required : get_option('use_libbsd'), allow_fallback: false)
+config.set('HAVE_ARC4RANDOM_BUF', cc.has_header_symbol('stdlib.h', 'arc4random_buf', dependencies: dep_libbsd))
+config.set('HAVE_ARC4RANDOM', cc.has_header_symbol('stdlib.h', 'arc4random', dependencies: dep_libbsd))
```

Comment:
Can't we just allow the dependency fallback, but if we do fallback then check if dep_libbsd.type_name() == 'internal' and unconditionally set these without running has_header_symbol at all?

Reply:
sounds like quite the hack :)

Reply:
Added this suggestion.